### PR TITLE
RMI-2469 Improve vault writing process

### PR DESCRIPTION
**Pull request by nixonmgr**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RMI-2469

### Change description ###
updated mi-set-secrets.yaml to directly take in secretValue

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by tchow8 — DISMISSED** [review]
Looks good

File: pipeline-steps/templates/steps/vault-upload.yaml [hunk 1]
```diff
@@ -15,12 +15,12 @@ steps:
             varValue='$(${{ pair.env_var_name }})'
```
Comment (qzhou-hmcts) [inline pipeline-steps/templates/steps/vault-upload.yaml:15]: change the name to var_value as it does not represent environment var name anymore

File: pipeline-steps/templates/steps/vault-upload.yaml [hunk 2]
```diff
@@ -12,15 +12,15 @@ steps:
           scriptLocation: "inlineScript"
           inlineScript: |
             seachedSecret=$(az keyvault secret list --vault-name ${{ parameters.azureVault }}  --query "[?name=='${{ pair.secret_name }}'].name" -o tsv)
-            varValue='$(${{ pair.env_var_name }})'
+            var_value='$(${{ pair.env_var_name }})'
```
Comment (tchow8) [inline pipeline-steps/templates/steps/vault-upload.yaml:15]: Sorry, I think Qiang meant the pair.env_var_name to pair.var_value.
The original variable name varValue is fine.

Comment (tchow8) [reply pipeline-steps/templates/steps/vault-upload.yaml:15]: Remember to also change it to match in the mi-set-secrets.yaml

Comment (nixonmgr) [reply pipeline-steps/templates/steps/vault-upload.yaml]: made the changes

File: pipeline-steps/templates/steps/vault-upload.yaml [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
-                 echo  "This is a new secret"
-                 az keyvault secret set --name ${{ pair.secret_name }} --vault-name ${{ parameters.azureVault }}  --value '$(${{ pair.env_var_name }})'
+  - task: AzureCLI@1
+    displayName: "Set vault secrets ${{ parameters.secretName }}"
+    inputs:
+      azureSubscription: ${{ parameters.azureSubscriptionEndpoint }}
+      scriptLocation: "inlineScript"
+      inlineScript: |
+        seachedSecret=$(az keyvault secret list --vault-name ${{ parameters.azureVault }}  --query "[?name=='${{ parameters.secretName }}'].name" -o tsv)
+        varValue='${{ parameters.secretValue }}'
```
Comment (tchow8) [inline pipeline-steps/templates/steps/vault-upload.yaml:15]: Very minor, but if you think you can do it quickly, try removing the usage of varValue and just directly using parameters.secretValue where varValue is referenced instead, or vice-versa.
If it doesn't work, then can leave as is.

**Review by qzhou-hmcts — APPROVED** [review]

**Pull request merged**